### PR TITLE
docs: Remove Welcome To Tilt page

### PR DESCRIPTION
**Pull request by lianmakesthings**
Hi @lizzthabet @SurbhiSGupta 

could you please review the following changes:
- removes all references to the "Welcome to Tilt" page
- removes the "Welcome to Tilt" page

### CODE REVIEW

File: src/_includes/header-docs.html [hunk 1]
```diff
@@ -3,7 +3,7 @@
 <header class="siteHeader2 siteHeader--docs">
   <div class="siteHeader-logoAndMenuButton">
     {% if page.layout == 'landing' %}
-    <!-- For the "Guide To App Devs" - /welcome_to_tilt.html -->
```
Comment (lianmakesthings) [inline src/_includes/header-docs.html:6]: I just removed this, so I can reference it in the PR 😅 
@lizzthabet what's the significance of this comment? Can we remove the following <a> tag as it won't be needed anywhere else?

**Review by SurbhiSGupta — APPROVED** [review]

File: src/_includes/header-docs.html (hunk 1 above)
Comment (hyu) [reply src/_includes/header-docs.html:6]: @lianmakesthings, thanks for noticing this defunct code. I made a separate [PR](https://github.com/tilt-dev/tilt.build/pull/1042) to clean this up, among other things.

Comment (lizzthabet) [reply src/_includes/header-docs.html:6]: thanks, @hyu and @lianmakesthings! i'm not very familiar with the history of that landing page, but i know i've been directed to it from other places before. this is all i can dig up atm: https://github.com/search?q=welcome_to_tilt+user%3Atilt-dev&type=code.

Comment (lizzthabet) [reply src/_includes/header-docs.html:6]: ...which as i'm looking through your pr, you've already handled! apologies for the unhelpful/premature comment. 😅 i can't see any reason to keep the layout block for a page we're not using anymore.

**Review by lizzthabet — APPROVED** [review]

**Review by hyu — APPROVED** [review]
Thanks for cleaning up this old page!

**Pull request merged**